internal/pkg/chunker/file/file.go

```diff
@@ -57,88 +56,97 @@ func NewChunker(source Source, setters ...Option) chunker.Chunker {
 }
 
 // Read implements ChunkReader.
+//
+// nolint: gocyclo
 func (c *File) Read(ctx context.Context) <-chan []byte {
 	// Create a buffered channel of length 1.
 	ch := make(chan []byte, 1)
 
 	go func(ch chan []byte) {
 		defer close(ch)
 
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			log.Printf("failed to watch: %v\n", err)
+			return
+		}
+		// nolint: errcheck
+		defer watcher.Close()
+
+		if err = watcher.Add(filepath.Dir(c.source.Name())); err != nil {
+			log.Printf("failed to watch add: %v\n", err)
+			return
+		}
 		offset, err := c.source.Seek(0, io.SeekStart)
 		if err != nil {
+			log.Printf("failed to seek: %v\n", err)
 			return
 		}
+
 		buf := make([]byte, c.options.Size)
+
 		for {
-			select {
-			case <-ctx.Done():
-				return
-			default:
+			for {
 				n, err := c.source.ReadAt(buf, offset)
-				if err != nil {
-					if err != io.EOF {
-						fmt.Printf("read error: %s\n", err.Error())
-						break
-					}
+				if err != nil && err != io.EOF {
+					log.Printf("read error: %s\n", err.Error())
+					return
 				}
+
 				offset += int64(n)
-				if n != 0 {
+
+				if n > 0 {
 					// Copy the buffer since we will modify it in the next loop.
 					b := make([]byte, n)
 					copy(b, buf[:n])
-					ch <- b
 
-					// Clear the buffer.
-					for i := 0; i < n; i++ {
```

Comment:
Is there any potential security risk by not clearing the buffer?

Reply:
I don't believe so, as we only ever return part of the buffer after read, so it should be fine, as it gets overwritten with new Read